packages/types/src/primitive/Extrinsic/Extrinsic.ts

```diff
@@ -31,12 +34,27 @@ const UNMASK_VERSION = 0b01111111;
  * - signed, to create a transaction
  * - left as is, to create an inherent
  */
-export default class Extrinsic extends Base<ExtrinsicV1> implements IExtrinsic, Codec {
-  public constructor (value?: ExtrinsicValue | AnyU8a | Method) {
-    super(Extrinsic.decodeExtrinsic(value));
+export default class Extrinsic extends Base<ExtrinsicV1 | ExtrinsicV2> implements IExtrinsic {
+  public constructor (value: Extrinsic | ExtrinsicValue | AnyU8a | Method | undefined, extrinsicVersion: number = DEFAULT_VERSION) {
```

Comment:
Could we have an object in 2nd arg here? `{ version?: number }`. Just in case if one day we want to pass meta too